gateway-ha/src/main/java/io/trino/gateway/ha/router/ExternalRoutingGroupSelector.java

```diff
@@ -76,7 +74,7 @@ public class ExternalRoutingGroupSelector
             throw new RuntimeException("Invalid URL provided, using "
                     + "routing group header as default.", e);
         }
-        httpClient = new JettyHttpClient(new HttpClientConfig());
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
```

Comment:
I would move this line to the 1st line in this method for consistency with parameter's order.